bin/improver-tests

```diff
@@ -18,46 +18,83 @@ function echo_ok {
     echo -e "\033[1;32m[OK]\033[0m $1"
 }
 
+function improver_test_pep8 {
+    # PEP8 testing.
+    ${PEP8:-pep8} improver
+    echo_ok "pep8"
+}
+
+function improver_test_pylint {
+    # Pylint obvious-errors-only testing.
+    ${PYLINT:-pylint} --rcfile=../etc/pylintrc improver
+}
+
+function improver_test_pylintE {
+    # Pylint obvious-errors-only testing.
+    ${PYLINT:-pylint} -E --rcfile=../etc/pylintrc improver
+    echo_ok "pylint -E"
+}
+
+function improver_test_unit {
+    # Unit tests.
+    python -m unittest discover
+    echo_ok "Unit tests"
+}
+
+function improver_test_cli {
+    # CLI testing.
+    PATH="$IMPROVER_DIR/tests/bin/:$PATH"
+    if [[ ${1:-} != '--debug' ]] && type prove &>/dev/null; then
+        prove -j $(nproc) -r -e "bats --tap" \
+            --ext ".bats" "$IMPROVER_DIR/tests/"
+    else
+        bats $(find "$IMPROVER_DIR/tests/" -name "*.bats")
+    fi
+    echo_ok "CLI tests"
+}
+
+cd $IMPROVER_DIR/lib
+
 if [[ ${1:-} == '--help' ]] || [[ ${1:-} == '-h' ]]; then
     cat <<'__USAGE__'
-improver tests [--debug]
+improver tests [--debug] [SUBTEST] 
 
 Run pep8, pylint, unit and CLI acceptance tests.
 
 Optional arguments:
     --debug         Run in verbose mode (may take longer for CLI)
     -h, --help          Show this message and exit
+
+Arguments:
+    SUBTEST         Name of a subtest to run without running the rest.
+                    Valid names are: pep8, pylint, pylintE, unit, cli.
+                    pep8, pylintE, unit, and cli are the default tests.
 __USAGE__
     exit 0
 fi
-if [[ -n "${1:-}" && ${1:-} != '--debug' ]]; then
+
+DEBUG_OPT=
+if [[ "${1:-}" == '--debug' ]]; then
+    DEBUG_OPT='--debug'
+    shift
+fi
+
+if [[ -n "${1:-}" ]]; then
+    TEST_NAME="improver_test_$1"
+    shift
+    if [[ $(type -t $TEST_NAME) == 'function' ]]; then
+        "$TEST_NAME" "$@" "$DEBUG_OPT"
+        exit 0
```

Comment:
This block effectively duplicates the testing instructions contained in the next code block with an `exit 0` used to prevent both blocks being used. It would be better to create a list of tests from command line options and if empty to use a default list.